java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JsonWebKeySet.java

```diff
@@ -7,17 +7,16 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import java.util.HashMap;
 import java.util.HashSet;
-import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 class JsonWebKeySet {
 
 	private final Set<JsonWebKey> jsonWebKeys = new HashSet<>();
-	private final Map<String, Boolean> appTidAccepted = new HashMap<>();
+	@Deprecated
+	private final Set<String> appTidAccepted = new HashSet<>();
```

Comment:
Yes, actually we can. The class itself is not public, so it is not an API breaking change if we do. Thanks!

Reply:
I think it was implemented this way to re-use objects of the class for multiple tenants that use the same keys, so that there are not multiple equal JsonWebTokenKey objects in memory.